### PR TITLE
Update nanoleaf documentation with examples

### PR DESCRIPTION
**Pull request by jay-tuckey**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @jay-tuckey, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#16537).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_integrations/nanoleaf.markdown [hunk 1]
```diff
@@ -24,6 +24,7 @@ light:
   - platform: nanoleaf
     host: 192.168.1.10
     token: xxxxxxxxxxxxxxxxxxxxx
+    name: bedroom_triangles
```
Comment (klaasnicolaas) [inline source/_integrations/nanoleaf.markdown]:
```suggestion
```
We always try to keep the first example as minimal as possible, so only the required options are used.

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_integrations/nanoleaf.markdown [hunk 2]
```diff
@@ -12,7 +12,7 @@ ha_platforms:
 
 ### Configuration Sample
 
-The `nanoleaf` platform allows you to control [Nanoleaf Light Panels](https://nanoleaf.me) from Home Assistant.
+The `nanoleaf` platform allows you to control [Nanoleaf Light Panels](https://nanoleaf.me) from Home Assistant. Note that for full control of nanoleaf devices, particularly effects, make sure they are set up as `- platform: nanoleaf` not as a HomeKit device.
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]:
```suggestion
The Nanoleaf integration allows you to control [Nanoleaf Light Panels](https://nanoleaf.me) from Home Assistant. Note that for full control of Nanoleaf devices, particularly effects, make sure they are set up as instructed on this page and not as a HomeKit device.
```

Additionally, I think the first sentence should be moved above the first heading.

File: source/_integrations/nanoleaf.markdown [hunk 3]
```diff
@@ -50,3 +51,58 @@ name:
...
 If you get a 403 Forbidden message, you probably did not press the *ON* button long enough. The time-frame to get a valid token is only 30 seconds, so you have to be quick to issue the curl request.
+
+### Getting useful values for automations
+
+First set up some effects with the Nanoleaf app, and have them installed on the device.
+
+Once installed, go to "Developer Tools" and find the state of the device, eg `light.bedroom_triangles`. You should end up with something like this:
+
+```yaml
+# Example nanoleaf state:
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]: Dropping states in documentation causes confusion in general, as people try to use them. Please remove this blob.

File: source/_integrations/nanoleaf.markdown [hunk 4]
```diff
@@ -50,3 +51,58 @@ name:
 4. The output should include the auth token like *{"auth_token":"xxxxxxxxxxxxxxxxxxxxx"}*, copy the resulting token into your configuration
 
 If you get a 403 Forbidden message, you probably did not press the *ON* button long enough. The time-frame to get a valid token is only 30 seconds, so you have to be quick to issue the curl request.
+
+### Getting useful values for automations
+
+First set up some effects with the Nanoleaf app, and have them installed on the device.
+
+Once installed, go to "Developer Tools" and find the state of the device, eg `light.bedroom_triangles`. You should end up with something like this:
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]: Why does one need the developer tools? The effects are available on the light entity itself, right?

```suggestion
Once installed, go to "Developer Tools" and find the state of the device, e.g., `light.bedroom_triangles`. You should end up with something like this:
```

File: source/_integrations/nanoleaf.markdown [hunk 5]
```diff
@@ -50,3 +51,58 @@ name:
...
+icon: 'mdi:triangle-outline'
+supported_features: 55
+```
+
+Now you can call the `light.turn_on` service with the data:
+```yaml
+# Example data to activate a nanoleaf effect:
+entity_id: light.trianglestest
+effect: Starlight
+```
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]: I feel this is out of place in this document. This is the integration documentation, not the documentation to teach how to use lights in Home Assistant.

While I get your goal, in theory, this example can be copied into hundreds of places, which is not maintainable.

File: source/_integrations/nanoleaf.markdown [hunk 6]
```diff
@@ -50,3 +51,58 @@ name:
...
+```yaml
+# Example alarm script
+alias: Alarm Sequence
+sequence:
+  - service: light.turn_on
+    data:
+      effect: Vibrant Sunrise
+    entity_id: light.triangles
+mode: single
+```
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]: same comment as above.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/nanoleaf.markdown (hunk 1 above)
Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: Ok, makes sense. I've removed the name from the first example, and made a second example with the name, but it's pretty obvious so I can drop it entirely if it's not adding any value.

File: source/_integrations/nanoleaf.markdown (hunk 3 above)
Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: Ok, can do. However when trying to set up my Nanoleaf I had quite a bit of confusing working out how to find the names of the effects that I could use and how to call them, I had to read the integration code just to work out what it was expecting. This is why I have made the examples of how to find the effects from the state and then call the service to set the lights later.

Is there a better way to communicate this? At the moment the page feels like "here's how to create the light entity" but with no info on how to actually use the entity that was created.

Comment (frenck) [reply source/_integrations/nanoleaf.markdown]: If you click on a light, it just shows them...

image

Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: Ok, that's good. Can I put a screenshot in this document? I'm not sure how I can do that.

Comment (frenck) [reply source/_integrations/nanoleaf.markdown]: Yes, however, the below concerns are in the same region at this one.

It is general Home Assistant functionality, and not specific to this integration.

File: source/_integrations/nanoleaf.markdown (hunk 5 above)
Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: Yeah, removed this snippet.

File: source/_integrations/nanoleaf.markdown (hunk 6 above)
Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: I feel this is worth keeping, although I've added some extra commenting explaining why it exists, showing how we can use the effects that we discover from the state/UI.

Comment (frenck) [reply source/_integrations/nanoleaf.markdown]: I don't agree. It is general light control, and not anything that is specific this integration. This does not belong on this page, as controlling a light is a generic thing, not a Nanoleaf thing.

Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: Ok, I have added a disclaimer noting the versions that were used for the examples, along with a link to the proper "Lights" integration page. This should direct users to the full docs for lights to find all the info that is available, but still give them something to get started with.

File: source/_integrations/nanoleaf.markdown [hunk 7]
```diff
@@ -50,3 +60,46 @@ name:
...
 
 If you get a 403 Forbidden message, you probably did not press the *ON* button long enough. The time-frame to get a valid token is only 30 seconds, so you have to be quick to issue the curl request.
+
+### Using the Nanoleaf Device
+
+The nanoleaf device is a standard light, so can be used with standard Home Assistant Light services for automations and scripts. For full details see the [Light Integrations Page.](/integrations/light/)
+
+### Getting Useful Values for Automations and Scripts
+
+These examples were written for version 2021.2 with a Nanoleaf Shapes device on version 5.0.0 of firmware, so newer versions may have introduced or changed functionality.
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]: We don't keep version history in our documentation. It always reflects the latest stable version of Home Assistant.

File: source/_integrations/nanoleaf.markdown [hunk 8]
```diff
@@ -50,3 +60,46 @@ name:
...
+# Example alarm script
+alias: Alarm Sequence
+sequence:
+  - service: light.turn_on
+    data:
+      # Note - this is one of the effects we discovered in the state
+      effect: Vibrant Sunrise
+    entity_id: light.bedroom_triangles
+mode: single
+```
```
Comment (frenck) [inline source/_integrations/nanoleaf.markdown]: This example should not be in this part of the documentation, as it is not specific or unique for Nanoleaf.

Comment (jay-tuckey) [reply source/_integrations/nanoleaf.markdown]: Ok, dropped all the examples.

**Review by frenck — APPROVED** [review]
Thanks, @jay-tuckey 👍

**Pull request merged**